### PR TITLE
fixed check ans screen , date issue , yesnoempty issues

### PR DESCRIPTION
**Pull request by SasankAchyuta9**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by channa-patil — APPROVED** [review]

**Review by AlokDatta — APPROVED** [review]

**Review by vivek-sekhar — APPROVED** [review]

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 1]
```diff
@@ -126,7 +161,10 @@ export const getProceedingDetails = (respondent: Respondent, req: AppRequest): P
 };
 
 function getLocalDate(orderDate: string): Date {
-  if (orderDate) {
+  if (orderDate['year'] === '' && orderDate['month'] === '' && orderDate['day'] === '') {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:164]: Also this if else block can be written much simpler with short syntax

File: src/main/steps/respondent/task-list/tasklist.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ const getYourResponseSection = (sectionTitles, taskListItems, userCase: CaseWith
             id: 'respond_to_allegations_of_harm_and_violence',
             text: taskListItems.respond_to_allegations_of_harm_and_violence,
             status: getInternationalFactorsStatus(userCase),
-            href: !hasCitizenResponse ? URL.INTERNATIONAL_FACTORS_START : null,
+            href: '#',
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/task-list/tasklist.ts:156]: Why is the href changed to #?

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 3]
```diff
@@ -126,7 +161,10 @@ export const getProceedingDetails = (respondent: Respondent, req: AppRequest): P
...
+    const formated_Date = new Date(orderDate[''], orderDate[''], orderDate['']);
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:165]: When we are not giving this condition.. what is happening means the order date object is taken as normal object with empty values in its properties , and which is leading to auto generation of date.

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts (hunk 1 above)
Comment (SasankAchyuta9) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:164]: Yes Vivek , This has been taken care .. the unnecessary else which was present before is now removed.

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**